lib/iolib/iolib.go

```diff
@@ -355,6 +357,55 @@ func open(t *rt.Thread, c *rt.GoCont) (rt.Cont, error) {
 	if ioErr != nil {
 		return pushingNextIoResult(t.Runtime, c, ioErr)
 	}
+
+	u := newFileUserData(f, getIoData(t.Runtime).metatable)
+	return c.PushingNext(t.Runtime, rt.UserDataValue(u)), nil
+}
+
+func popen(t *rt.Thread, c *rt.GoCont) (rt.Cont, error) {
+	if err := c.Check1Arg(); err != nil {
+		return nil, err
+	}
+	cmdStr, err := c.StringArg(0)
+	if err != nil {
+		return nil, err
+	}
+
+	mode := "r"
+	if c.NArgs() >= 2 {
+		mode, err = c.StringArg(1)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var cmdArgs []string
+	if runtime.GOOS == "windows" {
+		cmdArgs = []string{"C:\\Windows\\system32\\cmd.exe", "/c", cmdStr}
+	} else {
+		cmdArgs = []string{"/bin/sh", "-c", cmdStr}
+	}
+
+	cmd := exec.Cmd{
+		Path: cmdArgs[0],
+		Args: cmdArgs,
+	}
+
+	f := &File{}
+	switch mode {
+		case "r":
+			stdout, _ := cmd.StdoutPipe()
+			f.reader = &nobufReader{stdout}
+		case "w":
+			stdin, _ := cmd.StdinPipe()
+			f.writer = &nobufWriter{stdin}
+	}
```

Comment:
The mode should be validated and an error returned when it is invalid.  Running lua5.4 I get this:
```
> io.popen("echo foo", "a")
stdin:1: bad argument #2 to 'popen' (invalid mode)
```